### PR TITLE
fix(unittest): fix unittest data not initialized

### PR DESCRIPTION
**Pull request by ericlaz**
## Motivation

This PR aims to fix an issue with `unittest` which would cause CI Visibility data for `unittest` to be initialised incorrectly.

## Checklist

- [X] Change(s) are motivated and described in the PR description.
- [X] Testing strategy is described if automated tests are not included in the PR.
- [X] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [X] Change is maintainable (easy to change, telemetry, documentation).
- [X] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [X] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [X] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: ddtrace/contrib/unittest/patch.py [hunk 1]
```diff
@@ -51,6 +51,8 @@ def get_version():
 
 
 def _enable_unittest_if_not_started():
+    if not hasattr(_CIVisibility, "_unittest_data"):
+        _CIVisibility._unittest_data = {"suites": {}, "modules": {}}
```
Comment (brettlangdon) [inline ddtrace/contrib/unittest/patch.py:55]: is there a reason why we do the `hasattr` call here and add a new property dynamically vs defining `_unittest_data` on the `CIVisibility` in the class definition?

Comment (romainkomorndatadog) [reply ddtrace/contrib/unittest/patch.py:55]: `CIVIsibility` isn't inherently tied to `unittest`, so I don't particularly like the idea of having test-framework-related data in there. Putting it in the class definition "codifies" it in a way I don't like.

Doing it this way has its own janky vibe, and very much "half a dozen of one, six of the other".

Either way, I plan to rewrite `CIVisibility` somewhat extensively in Q4 or Q1 anyway, so whatever change we make here will be relatively short-lived.

Comment (brettlangdon) [reply ddtrace/contrib/unittest/patch.py:55]: 👍🏻

File: releasenotes/notes/fix-unittest-data-not-initialized-4c08983165adcfc1.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    CI Visibility: fixes `unittest` data not being initialized properly
```
Comment (brettlangdon) [inline releasenotes/notes/fix-unittest-data-not-initialized-4c08983165adcfc1.yaml:4]:
```suggestion
    CI Visibility: fixes ``unittest`` data not being initialized properly
```

**Review by brettlangdon — DISMISSED** [review]

**Review by majorgreys — APPROVED** [review]

**Pull request merged**